### PR TITLE
Classes and Interface added.

### PR DESCRIPTION
**Pull request by ritakisluhina**

### CODE REVIEW

File: src/main/java/core/basesyntax/FigureGenerator.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureGenerator {
+    private static final String[] arrayOfFigures = new String[]{
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/FigureGenerator.java:6]: i don't usage of this variable in the code, do we need it somewhere?

File: src/main/java/core/basesyntax/FigureGenerator.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            "square", "rectangle", "right triangle", "circle", "isosceles trapezoid"};
+
+    public static Figure createFigure() {
+        switch (new Random().nextInt(5)) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/FigureGenerator.java:10]: 5 is a magic number, let's declare it as a constant

File: src/main/java/core/basesyntax/FigureGenerator.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            case 5:
+                return new IsoscelesTrapezoid(getRandomNumber(),
+                        getRandomNumber(), getRandomNumber(), getRandomColor());
+            default:
+                return null;
+        }
+    }
+
+    private static String getRandomColor() {
+        int randomGenerateIndex = new Random().nextInt(6);
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/FigureGenerator.java:29]: if you have six colors (enum length) it's better not to hardcode this value 
sue - Color.values().length

File: src/main/java/core/basesyntax/RightTriangle.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+public class RightTriangle extends Shape implements Figure {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/RightTriangle.java:3]: we don't need to implement Figure here, cause Shape already did it

File: src/main/java/core/basesyntax/Main.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+import java.util.Random;
+
+public class Main {
+    public static void main(String[] args) {
+        FigureGenerator[] randomArray = new FigureGenerator[new Random().nextInt(10)];
+        for (int i = 0; i < randomArray.length; i++) {
+            randomArray[i] = (FigureGenerator) FigureGenerator.createFigure();
+        }
+        for (FigureGenerator print : randomArray) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Main.java:11]:
```suggestion
        for (FigureGenerator figure : randomFiguresArray) {
```
let's improve a naming a bit

File: src/main/java/core/basesyntax/Figure.java [hunk 6]
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public interface Figure {
+    public void draw();
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Figure.java:4]: methods in interfaces are public abstract by default

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/FigureGenerator.java (hunk 1 above)
Comment (ritakisluhina) [reply src/main/java/core/basesyntax/FigureGenerator.java]: done

File: src/main/java/core/basesyntax/Figure.java [hunk 7]
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public interface Figure {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Figure.java:3]:
```suggestion
public interface Drawable {
```

File: src/main/java/core/basesyntax/Shape.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+
+    public double getArea() {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Shape.java:19]: you override this method in each class, so let's create a new interface and move this method there, and then remove field area from the class

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Circle.java [hunk 9]
```diff
@@ -0,0 +1,26 @@
+package core.basesyntax;
+
+public class Circle extends Shape {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Circle.java:3]: add getRadius method as well

Comment (ritakisluhina) [reply src/main/java/core/basesyntax/Circle.java:3]: done

File: src/main/java/core/basesyntax/IsoscelesTrapezoid.java [hunk 10]
```diff
@@ -1,6 +1,6 @@
 package core.basesyntax;
 
-public class IsoscelesTrapezoid extends Shape implements Figure {
+public class IsoscelesTrapezoid extends Shape implements Drawable {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/IsoscelesTrapezoid.java:3]: you don't need to implement Drawable here, cuase Shape already does

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request closed without merging**